tests/locallib_test.php

```diff
@@ -28,16 +28,121 @@
 global $CFG;
 require_once($CFG->dirroot . '/mod/accredible/locallib.php');
 
+use mod_accredible\apiRest\apiRest;
+
 class mod_accredible_locallib_testcase extends advanced_testcase {
     /**
-     * Sample test
-     * later: remove after adding real tests
+     * Setup before every test.
      */
-    public function test_sample() {
+    public function setUp(): void {
         $this->resetAfterTest();
         $this->setAdminUser();
-    
-        // Sample test.
-        $this->assertEmpty("");
+
+        // Add plugin settings.
+        set_config('accredible_api_key', 'sometestapikey');
+        set_config('is_eu', 0);
+
+        $this->mockapi = new class {
+            /**
+             * Returns a mock API response based on the fixture json.
+             * @param string $jsonpath
+             * @return array
+             */
+            public function resdata($jsonpath) {
+                global $CFG;
+                $fixturedir = $CFG->dirroot . '/mod/accredible/tests/fixtures/mockapi/v1/';
+                $filepath = $fixturedir . $jsonpath;
+                return json_decode(file_get_contents($filepath));
+            }
+        };
+    }
+
+    /**
+     * Test whether it returns group name arrays
+     */
+    public function test_accredible_get_templates() {
+        /**
+         * When the apiRest returns groups.
+         */
+        $mockclient1 = $this->getMockBuilder('client')
+                            ->setMethods(['post'])
+                            ->getMock();
+
+        // Mock API response data.
+        $resdata = $this->mockapi->resdata('groups/search_success.json');
+
+        $reqdata = json_encode(array('page' => 1, 'page_size' => 10000));
+
+        // Expect to call the endpoint once with page and page_size.
+        $url = 'https://api.accredible.com/v1/issuer/groups/search';
+        $mockclient1->expects($this->once())
+                    ->method('post')
+                    ->with($this->equalTo($url),
+                           $this->equalTo($reqdata),)
+                    ->willReturn($resdata);
```

Comment:
The test fails here as the `$url `doesn't match the one from `$this->api_endpoint`.
This is because the `'ACCREDIBLE_DEV_API_ENDPOINT'` environment variable is not set to empty in `locallib_test`

Reply:
It worked in my local Moodle for some reason. I added a line to unset the environment variable in `setUp()` in locallib_test.php. Thanks!